File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cpp
class Solution {
public:
    void postorderTraversalHelper(TreeNode* currentNode, vector<int>& result) {
        if (currentNode == nullptr) return;
        postorderTraversalHelper(currentNode->left, result);
        postorderTraversalHelper(currentNode->right, result);
        result.push_back(currentNode->val);
    }

    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> result;
        postorderTraversalHelper(root, result);
        return result;
    }
};

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
class Solution {
    private void postorderTraversalHelper(TreeNode currentNode, List<Integer> result) {
        if (currentNode == null) {
            return;
        }
        postorderTraversalHelper(currentNode.left, result);
        postorderTraversalHelper(currentNode.right, result);
        result.add(currentNode.val);
    }

    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        postorderTraversalHelper(root, result);
        return result;
    }
}